src/soroban_rpc.ts

```diff
@@ -36,6 +37,8 @@ export namespace SorobanRpc {
     key: string;
     /** a base-64 encoded {@link xdr.LedgerEntryData} instance */
     xdr: string;
+    /** an optional base-64 endoded  {@link xdr.ExpirationEntry} instance*/
+    expiration?: string;
```

Comment:
that way it passes through parseRawLedgerEntries seamless to include it on `GetLedgerEntriesResponse`, represents the model change we'd mostly anticipate from rpc to follow, 'contract first' approach?

Reply:
I see what you're saying but this busts the `_getLedgerEntries` schema that returns the RPC directly, right? Including this in the non-raw / parsed version makes more sense to me. And we can modify this once the RPC is updated.

Reply:
I don't understand how it would break compat, `expiration` is optional, current rpc responses which don't have `expiration` shouldn't trigger runtime error when marshaled into this, and `mergeResponseExpirationLedgers` takes in this raw response model from rpc first, populates `expiration` on the model if possible, and forwards to parse method, ideally https://github.com/stellar/soroban-tools/issues/1010 will just update the response model to same, then it's just small change here in client to remove `mergeResponseExpirationLedgers`/`expandRequestIncludeExpirationLedgers`, the rest of this should remain and work the same.

Reply:
ah, well, have to change the tests also to no longer expect axiosMock to include expiration keys in requests and expiration entry in responses.

Reply:
I'm going to replace `expiration` with `expirationLedgerSeq` on `RawLedgetEntryResult.expiration` and `LegderEntryResult` per [the review comment](https://github.com/stellar/js-soroban-client/pull/161/files/6cda5badbdef651d8a0da50f51e0b929c3634ffa#r1353836257) which also lines up with what was mentioned for specs on [rpc side/#1010](https://github.com/stellar/soroban-tools/issues/1010), will have the same optional flow here in client for the interim, but ideally client will have small, low-impact cut-over path once present on rpc responses.

Reply:
okay I see your rationale! I wasn't thinking about marshalling, more-so that the `Raw*` schema should line up exactly with the RPC schema. But since we're getting #1010 out soon, and since the field is optional, I'm fine with including here in advance.